parquet-hadoop/src/main/java/org/apache/parquet/hadoop/InternalParquetRecordReader.java

```diff
@@ -257,4 +259,64 @@ public boolean nextKeyValue() throws IOException, InterruptedException {
     return Collections.unmodifiableMap(setMultiMap);
   }
 
+  //TODO maintain unmaterializableRecordCounter properly
+  public boolean nextBatch(ObjectColumnVector<T> vector) throws IOException {
```

Comment:
Yep this is a wrapper around the regular record reader and `ParquetReader::readVector(ObjectColumnVector<T> vector)` uses this method (as it only has access to the `InternalParquetRecordReader`, not to the underlying `RecordReader`. Do you have a suggestion?